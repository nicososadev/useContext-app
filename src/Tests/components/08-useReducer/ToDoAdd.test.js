import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { ToDoAdd } from '../../../components/08-useReducer/ToDoAdd'

describe('Pruebas en el componente ToDoAdd', () => {

    const handleAdd = jest.fn()

    const wrapper = shallow( 
        <ToDoAdd 
            handleAdd={ handleAdd }
        /> 
    )

    test('Debe mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('No debe llamar la funcion handleAdd', () => {

        const formSubmit = wrapper.find('form').prop('onSubmit')

        formSubmit({ preventDefault(){} })

        expect( handleAdd ).not.toHaveBeenCalled()
    })
    
    test('Debe llamar la funcion handleAdd', () => {

        const value = 'Tarea'

        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        })

        const formSubmit = wrapper.find('form').prop('onSubmit')

        formSubmit({ preventDefault(){} })

        expect( handleAdd ).toHaveBeenCalled()
        expect( handleAdd ).toHaveBeenCalledWith( expect.any(Object) )
        expect( handleAdd ).toHaveBeenCalledWith({
            desc: value,
            done: false,
            id: expect.any(Number)
        })

        expect( wrapper.find('input').prop('value') ).toBe( '' )
    })
})
