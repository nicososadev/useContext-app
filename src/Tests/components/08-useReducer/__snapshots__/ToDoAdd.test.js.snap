// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ToDoAdd Debe mostrarse correctamente  1`] = `
<Fragment>
  <h4>
    Agregar ToDo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Nueva Tarea"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-3 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
